### PR TITLE
Fix settings_dict

### PR DESCRIPTION
**Pull request by wanliAlex**
* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the current behavior?** (You can also link to an open issue here)
if we provided `settings_dict` with other parameters when creating index, there is no error out

* **What is the new behavior (if this is a feature change)?**
We error out if settings_dict is provided with other parameters in create_index

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Other information**:

### CODE REVIEW

**Review by papa99do — DISMISSED** [review]
LGTM

File: tests/v2_tests/test_create_index.py [hunk 1]
```diff
@@ -424,4 +433,116 @@ def test_create_structured_index_with_languagebind(self):
...
+        cls.get_index_response: List[Dict] = requests.get(
+            f"{cls.authorized_url}/api/v2/indexes",
+            headers={"x-api-key": os.environ.get("MARQO_API_KEY", None)}
+        ).json()["results"]
+
+    def test_cloud_index_attributes(self):
+        test_indexes = list(index_name_to_settings_mappings.keys())
+        for test_index in test_indexes:
+            index_name = test_index + "_" + os.environ.get("MQ_TEST_RUN_IDENTIFIER", "")
+            with self.subTest(f"Index name: {index_name}"):
```
Comment (papa99do) [inline tests/v2_tests/test_create_index.py:536]: Why do we need to test each Cloud index? Will it need to create a cloud index for each sub test? How long would this test take?

Comment (wanliAlex) [reply tests/v2_tests/test_create_index.py:536]: No, this test does not create any indexes. It only tests on the created indexes.

**Review by papa99do — APPROVED** [review]
LGTM

**Pull request merged**